src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs

```diff
@@ -546,13 +607,17 @@ internal static void ToggleItemSelectionAtCoerced(ItemsRepeater ir, int index)
 			SetIsSynchronizingSelection(ir, false);
 		}
 	}
-	internal static void SetItemSelection(DependencyObject x, bool value)
+	internal static void SetItemSelection(ItemsRepeater ir, DependencyObject itemRoot, bool value)
 	{
-		if (x is SelectorItem si)
+		var host = GetUseNestedSelectionHost(ir)
```

Comment:
Could we maybe avoid needing to have both a `UseNestedSelectionHost` and a `IsSelectionHost` property? Could we get away with just `IsSelectionHost`?

Perhaps just search for a selection host if the `itemRoot` itself is anything other than a `SelectorItem` or `ToggleButton`, otherwise assume the root should be the selection host? Or are there common scenarios that this logic wouldn't support?